common/lib/xmodule/xmodule/js/spec/problem/edit_spec.coffee

```diff
@@ -597,66 +598,22 @@ describe 'MarkdownEditingDescriptor', ->
 
 
     </problem>""")
-    it 'tests malformed labels', ->
-      data = MarkdownEditingDescriptor.markdownToXml("""
-        France is a country in Europe.
-
-        >>What is the capital of France?<
-        = Paris
-
-        blah>>What is the capital of <<Germany?<<
-        ( ) Bonn
-        ( ) Hamburg
-        (x) Berlin
-        ( ) Donut
-      """)
-      expect(data).toEqual("""<problem>
-    <p>France is a country in Europe.</p>
-
-    <p>>>What is the capital of France?<</p>
-    <stringresponse answer="Paris" type="ci" >
-      <textline size="20"/>
-    </stringresponse>
-
-    <p>blahWhat is the capital of Germany?</p>
-    <multiplechoiceresponse>
-      <choicegroup label="What is the capital of &lt;&lt;Germany?" type="MultipleChoice">
-        <choice correct="false">Bonn</choice>
-        <choice correct="false">Hamburg</choice>
-        <choice correct="true">Berlin</choice>
-        <choice correct="false">Donut</choice>
-      </choicegroup>
-    </multiplechoiceresponse>
-
 
-    </problem>""")
     it 'adds labels to formulae', ->
       data = MarkdownEditingDescriptor.markdownToXml("""
       >>Enter the numerical value of Pi:<<
       = 3.14159 +- .02
       """)
-      expect(data).toEqual("""<problem>
-    <p>Enter the numerical value of Pi:</p>
-    <numericalresponse answer="3.14159">
-      <responseparam type="tolerance" default=".02" />
-      <formulaequationinput label="Enter the numerical value of Pi:" />
-    </numericalresponse>
-
+      expect(data).toXMLEqual("""<problem>
+        <numericalresponse answer="3.14159">
+          <label>Enter the numerical value of Pi:</label>
+        <responseparam type="tolerance" default=".02"/>
+          <formulaequationinput label="Enter the numerical value of Pi:"/>
+        </numericalresponse>
 
-    </problem>""")
-    it 'escapes entities in labels', ->
```

Comment:
Removed because neither markdown nor the xml in above test case is rendered correctly causing the problem to raise exceptions at backend. I have tested it manually by creating the problem on stage with above markdown and xml.

Just found that the escaping is not implemented correctly which is tested in the above test. I will fix it and rewrite the test.